UN-OCHA/rwint9-site
# [RW-805] Add tracking parameters to URLs in notification emails

Refs: RW-805

This adds link tracking to the RW subscriptions.

It can be enabled/disabled per subscription via drush commands.

## Tests

**Prerequisite:** use the local stack from the repo and make sure `mail.rwint-local.test` is in your `/etc/hosts`.

Lots of bullet points below but the test is actually relatively straightforward.

1. Checkout the branch, clear the cache
2. Add a temporary `$config['reliefweb_api.settings']['api_url'] = 'https://api.reliefweb.int/v1';` at the bottom of the `settings.site.php`, so that the subscriptions uses the latest content from RW
3. Run `drush sql-query "truncate reliefweb_subscriptions_subscriptions"` to remove all the existing subscriptions
4. Run `drush sql-query "truncate reliefweb_subscriptions_queue"` to remove all the existing queued subscriptions
5. Run `drush sql-query "truncate reliefweb_subscriptions_logs"` to clear the logs about the last time subscriptions were sent
6. Run `drush -v reliefweb_subscriptions:enable-link-tracking headlines` to enable tracking for the headlines subscription. Check the output to confirm that everything is `off` except for `Link tracking for "ReliefWeb Headlines": on`
7. Log in as an admin
8. Go to `/admin/people` and click the name of the first user in the list and then its `Subscriptions` tab.
9. Check the `headlines` and `appeals` subscriptions.
10. Go to `/admin/subscriptions/overview`
11. Click the preview for `headlines` and check some links to confirm they have the following parameters: `utm_source=rw-subscriptions&utm_medium=email&utm_campaign=headlines`.
12. Go back to `/admin/subscriptions/overview`
13. Click the preview for `appeals` and confirm the links **do not have** the `utm` parameters (tracking was not enabled for this subscription)
14. Run `drush relieweb_subscriptions:queue headlines`
15. Run `drush reliefweb_subscriptions:send`
16. Run `drush relieweb_subscriptions:queue appeals`
17. Run `drush reliefweb_subscriptions:send`
18. Go to `mail.rwint-local.test` (assuming the use of the repo local stack, make sure this domain is in your `/etc/hosts`)
19. There should 1 email for the headlines and 1 for the appeals
20. Open them and confirm the links in the `headlines` including the `unsubscribe` at the bottom have the `utm` parameters and that the links for the `appeals` email don't have them.
21. Copy the `unsubscribe` link in the `headlines` email and open it in an incognito tab and confirm it proposes to unsubscribe from the headlines.
22. Remove the temporary setting from (2).
